src/components/layouts/CharactersLayout/CharactersLayout.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import {
+  ActivityIndicator,
+  View,
+  StyleSheet,
+  TouchableOpacity,
+} from "react-native";
+import { FlatList } from "react-native-gesture-handler";
+
+import Header from "../../Header/Header";
+import useMarvelData from "../../../hooks/ApiHelperHook";
+import CharacterCard from "../../Cards/CharacterCard";
+import theme from "../../../styles/theme/theme";
+import { useNavigation } from "@react-navigation/native";
+import routes from "../../../navigation/routes";
+import StatusIndicator from "../../StatusIndicator";
+import { useTranslation } from "react-i18next";
+
+export default function CharactersLayout({ endpoint }) {
+  const { data: characters, error, isLoading } = useMarvelData(endpoint);
+  const navigation = useNavigation();
+  const { t } = useTranslation();
+
+  function handleNavigation(item) {
+    navigation.navigate(routes.DETAIL_SCREEN, { item, type: "characters" });
+  }
+  function renderItem({ item }) {
```

Comment:
Arrow function kendi içinde context barındırmadığı için daha lightweight davranır. renderItem için o tercih edilmeli.